File: equations/fibonacci.py
# See https://en.wikipedia.org/wiki/Fibonacci_number

from sympy import fibonacci

if __name__ == "__main__":
    cases = [
        "fibonacci(0)",
        "fibonacci(1)",
        "fibonacci(6)",
        "fibonacci(91)",
        "fibonacci(283)",
        "fibonacci(3674)",
        "fibonacci(29821)"
    ]

    for lhs in cases:
        print("%s = %s" % (lhs, eval(lhs)))
    print("")

File: eyelet/input/fibonacci.pl
% See https://en.wikipedia.org/wiki/Fibonacci_number

:- op(1200, xfx, :+).

fibonacci(A, B) :-
    fibonacci(A, 0, 1, B).

fibonacci(0, A, _, A).
fibonacci(1, _, A, A).
fibonacci(A, B, C, D) :-
    A > 1,
    E is A-1,
    F is B+C,
    fibonacci(E, C, F, D).

golden_ratio(A, B) :-
    fibonacci(A, C),
    D is A+1,
    fibonacci(D, E),
    B is E/C.

% query
true :+ fibonacci(1, _).
true :+ fibonacci(2, _).
true :+ fibonacci(3, _).
true :+ fibonacci(4, _).
true :+ fibonacci(5, _).
true :+ fibonacci(91, _).
true :+ fibonacci(283, _).
true :+ fibonacci(3674, _).
true :+ golden_ratio(1, _).
true :+ golden_ratio(10, _).
true :+ golden_ratio(100, _).
true :+ golden_ratio(1000, _).
